spec/my_map_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require_relative '../lib/my_enumerables'
+
+RSpec.describe Enumerable do
+  subject(:enumerable) { [1, 1, 2, 3, 5, 8, 13, 21, 34] }
+
+  describe '#my_map' do
+    let(:double_value) do
+      proc { |value| value * 2 }
+    end
+
+    # Start by running these tests where map only needs to take a block
+    context 'when given a block' do
+      it 'returns an array by yielding to the block' do
+        expect(enumerable.my_map { |value| value * 2 }).to eq([2, 2, 4, 6, 10, 16, 26, 42, 68])
+      end
+
+      it 'returns an array with the same size as the enumerable' do
+        expect(enumerable.my_map { |value| value * 2 }.size).to eq enumerable.size
+      end
+    end
+
+    # Update map to take a proc instead.
+    # Then, unskip these tests by changing xcontext to context
+    # and skip the 'when given a block' tests by changing context to xcontext
+    xcontext 'when given a proc' do
+      it 'returns an array by calling the proc' do
+        expect(enumerable.my_map(double_value)).to eq([2, 2, 4, 6, 10, 16, 26, 42, 68])
```

Comment:
If we try to pass a proc into the original map method it will raise an argument error, I think we should keep it as close to the original map behaviour as we can incase there is any confusion.

We could instead test for something like this:
```
expect(enumerable.my_map(&:to_f)).to eq([1.0, 1.0, 2.0, 3.0, 5.0, 8.0, 13.0, 21.0, 34.0])
```

Reply:
I agree, I had those tests there because that was in the original requirements.

Reply:
Do you think there is any redundancy in testing passing in &:method since it's ruby that is responsible for converting the symbol to a proc and converting the proc to a block?

We would be essentially testing the same thing as just passing in a block. Though I do think it'd be a good thing to show to learners.